File: leetcode/0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.swift
class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        var i = 0;
        while i < nums.count - 1 {
            if(nums[i] == nums[i + 1]) {
                nums.remove(at: i);
            } else {
                i += 1
            }
        }

        return nums.count
    }
}

File: leetcode/0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int k = 0;
        int curIndex = 0;
        int curValue = -101;

        for(int i = 0; i < nums.size(); i++) {
            if(curValue < nums[i]) {
                nums[curIndex] = nums[i];
                curValue = nums[i];
                curIndex++;
                k++;
            }
        }

        return k;
    }
};